src/de/mod.rs

```diff
@@ -45,13 +47,17 @@ impl<'de, R: BincodeRead<'de>, O: Options> Deserializer<R, O> {
     }
 
     fn read_vec(&mut self) -> Result<Vec<u8>> {
-        let len: usize = serde::Deserialize::deserialize(&mut *self)?;
-        self.read_bytes(len as u64)?;
+        let len = O::ArraySize::read(&mut || serde::Deserialize::deserialize(&mut *self))?;
```

Comment:
Can we avoid shadow names ? (Newbie comment  ;-) )

Reply:
Here `len` isn't actually a shadow name. 
When the code does `len.try_into()` the method `try_into()` takes ownership of `len` so the variable ceases to exist. If I were to write:
```
        let len2: usize = len.try_into().map_err(|_e| ErrorKind::SizeLimit)?;
        println!("{}", len);
```
It actually wouldn't compile because there is no such thing as `len` anymore. It was consumed by `try_into()`. So using the variable is neither ambiguous or hiding any other variable. 

I do agree that it is bad form to reuse a variable name for something unrelated within the same method. However it's actually a common pattern in Rust to use the same name for an object that has gone through a transformation like this. In this case `len` was originally a `u64` and now is a `usize`, with the same value and meaning. (The case where the value does not fit into a `usize` results in an error in which case `len` is never assigned.)